src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -69,6 +71,7 @@ export function createEuiListItem({
     onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
       onClick();
       if (
+        !externalLink && // ignore external links
```

Comment:
NIT: by using the new `navigateToUrl` API instead of `navigateToApp`, we could get rid of this additional check and `externalLink` parameter. But platform can do it later.